.github/workflows/test-plugin.yml

```diff
@@ -97,3 +97,88 @@ jobs:
     - name: Run CPU backend examples
       working-directory: triton/third_party/triton_shared/python/examples
       run: pytest .
+
+
+
+  build_and_test_triton_shared_arm:
+    runs-on: ["self-hosted", "1ES.Pool=triton-shared-github-arm"]
+
+    steps:
+    - name: Check Architecture
+      run: uname -a
+
+    - name: Force Failure
+      if: ${{ inputs.force-failure }}
+      run: exit 1
+
+    - name: Checkout Triton
+      uses: actions/checkout@v4
+      with:
+        repository: 'openai/triton'
+        ref: ${{ inputs.triton-ref }}
+        path: triton
+        submodules: 'recursive'
+
+    - name: Checkout Triton-Shared
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.triton-shared-ref }}
+        path: triton/third_party/triton_shared
+
+    - name: Clear Triton Cache
+      run: |
+        rm -rf ~/.triton
+
+    - name: install pip
```

Comment:
i presume this is an ARM thing... i'll follow to see why this is missing

Reply:
@NathanielMcVicar following up :)
how does this look to you?

Reply:
I don't think it's an ARM thing, it's that the self-hosted runners come with a bunch of software pre-installed that's not in the base image, so having to install things as part of the run when using the Marketplace image is expected. We could also use Docker if the install time got out of hand, but for now I think just installing it in the yml (or a setup .sh script) is the best approach.

Reply:
I think since the runner is self-hosted and perhaps pip wasn't installed so I added that and was probably going to delete it after it ran with it once.

Reply:
These runners aren't self-hosted in the sense you mean here. These are hosted by the Microsoft 1ES infrastructure (see here https://github.com/apps/1es-resource-management) and work mostly just like the github hosted runners. In particular, they are stateless, so anything like installing pip you will have to do on every run.

Reply:
![image](https://github.com/microsoft/triton-shared/assets/48022006/06ce5534-1a60-47ae-9fa0-88d14c65b018)
https://github.com/microsoft/triton-shared/actions/runs/7067334608/job/19308128182#step:10:91

looks like it doesn't have a C++ compiler either, is it just build-essential that needs to be installed?